src/main.js

```diff
@@ -235,17 +203,98 @@ async function startFromFile(configPath, rulesPath) {
       console.log('Changes detected. Synchronizing...');
       const maps = rowsToObjects(rows);
 
-      const quads = await objectsToRdf({data: maps}, rml);
-
-      const deletedQuads = onlyInLeft(previousQuads, quads, compareQuads);
-      const addedQuads = onlyInLeft(quads, previousQuads, compareQuads);
-      previousQuads = quads;
+      const quads = await objectsToRdf(config, {data: maps}, rml);
 
-      await updateResource(deletedQuads, addedQuads, config.source);
+      const deletedQuads = config.diffChecker(previousData, quads, compareQuads);
+      const addedQuads = config.diffChecker(quads, previousData, compareQuads);
+      previousData = quads;
+      await config.resourceUpdater(deletedQuads, addedQuads);
     }
   }, config.interval);
 }
 
+/**
+ * @param {string} host - host of the resource, used to gather websocket endpoints
+ * @param {string} src - resource url
+ * @returns {Promise<boolean>} - true when using websockets, false when using a timer.
+ */
+async function setupResourceListening(host, src) {
+  // Pod -> Sheet sync
+  const websocketEndpoints = await getNotificationChannelTypes(host + '/.well-known/solid');
+  if (websocketEndpoints.length > 0 && websocketEndpoints[0].length > 0 && (!config.noWebsockets)) {
+    // listen using websockets
+    const url = websocketEndpoints[0];
+    const ws = await getWebsocket(url, src);
+    ws.on('message', async (notification) => {
+      const content = JSON.parse(notification);
+      if (content.type === 'Update') {
+        const res = await updateSheet();
+        if (!res) {
+          console.log(`[${src}]\tgot notified but the latest changes are already present`);
+        }
+      }
+    });
+    return true;
+  }
+  return false;
+}
+
+/**
+ * Queries resource and writes changes to sheet
+ * @returns {Promise<boolean>} true if there was new data written to the sheet, false if not
+ */
+async function updateSheet() {
+  const {results} = await queryResource(config, true);
```

Comment:
I would rename `queryResource` to `queryResources`, because now it might lead to confusing if this is only for when using one resource.